### PR TITLE
removed variable shadowing in 'hack'

### PR DESCRIPTION
**Pull request by stevekuznetsov**
@smarterclayton PTAL

blocked on https://github.com/openshift/origin/pull/7586

### CODE REVIEW

File: hack/jsonformat.go [hunk 1]
```diff
@@ -13,8 +13,8 @@ func main() {
 		log.Fatal("Usage: jsonformat.go <filename>")
 	}
 
-	byt, err := ioutil.ReadFile(os.Args[1])
-	if err != nil {
+	byt, readErr := ioutil.ReadFile(os.Args[1])
+	if readErr != nil {
 		log.Fatalf("ERROR: Unable to read file: %v\n", os.Args[1])
```
Comment (deads2k) [inline hack/jsonformat.go]: While you're here, give me the error.

File: hack/jsonformat.go [hunk 2]
```diff
@@ -13,8 +13,8 @@ func main() {
 		log.Fatal("Usage: jsonformat.go <filename>")
 	}
 
-	byt, err := ioutil.ReadFile(os.Args[1])
-	if err != nil {
+	byt, readErr := ioutil.ReadFile(os.Args[1])
```
Comment (smarterclayton) [inline hack/jsonformat.go]: Why is this shadowing.  Redefining err is not an anti pattern in the same level.  This is anti our conventions.

Comment (stevekuznetsov) [inline hack/jsonformat.go]: This error was being shadowed by another error later, and I chose to rename this one and not the one that was aliasing it.

File: hack/jsonformat.go (hunk 1 above)
Comment (stevekuznetsov) [inline hack/jsonformat.go]: I kind of wanted to keep these pulls clean and shadow-related only. I'll add it in here but I'm not sure that sort of "small edit" is going to scale to larger packages with 30 shadowing instances.

File: hack/verify-govet.sh [hunk 3]
```diff
@@ -56,7 +56,8 @@ DIR_BLACKLIST='./hack
 ./pkg/user
 ./pkg/util
 ./test
-./tools'
+./tools
+'
```
Comment (stevekuznetsov) [inline hack/verify-govet.sh]: This is going to keep other delete diffs clean if they touch the end of the list

**Comment by stevekuznetsov** [comment]
@smarterclayton we run into an issue. Some of the weirdness in this pull comes from a bug in `go tool vet` that isn't flagging the second `if` statement in `jsonformat.go` as shadowing, but nevertheless we have the issue of forced shadowing when using `:=` assignment inside of an `if` statement. 

The `if` token begins a new scope, so doing a `if b, err := someFunc()` won't re-assign `err`, even if `err` exists in the outer scope. Similarly. `if err := someFunc()` will declare a new `err` in the `if` block, which masks that which is outside. For instance:
```go
func main() {
    a, err := someFunc()
    if err != nil {
        return
    }

    if err := someOtherFunc(); err != nil { // this `err` is a new declaration and masks the outer `err`
        return
    }

    if b, err := someThirdFunc(); err != nil { // this `err` is a new declaration and masks the outer `err`
        return
    }
}
```

*Even if* we do 
```go
func main() {
    var theOneAndOnly error
    a, theOneAndOnly := someFunc()
    if err != nil {
        return
    }

    if theOneAndOnly := someOtherFunc(); err != nil { // this `theOneAndOnly` is a new declaration and masks the outer `theOneAndOnly` as `:=` can't re-assign if it's not a multi-variate assignment
        return
    }

    if b, theOneAndOnly := someThirdFunc(); err != nil { // this `theOneAndOnly` is a new declaration and masks the outer `theOneAndOnly`, although `:=` is mult-variate here, the previous `theOneAndOnly` wasn't declared in the same block and therefore cannot be re-assigned
        return
    }
}
```

So either we're going to need to bring things outside of the `if` statements, which is not idiomatic, or we're going to need to name things differently, which is also not idiomatic. Pick your poison.

**Comment by danmcp** [comment]
@stevekuznetsov Needs rebase
@smarterclayton Bump

**Comment by stevekuznetsov** [comment]
@smarterclayton we should sit down and hash out the best way to approach this ... the *few* cases where it's not idiomatic far outweigh the many cases where we benefit from the shadow check

**Pull request closed without merging**